obs_common/waitfor.py

```diff
@@ -47,18 +68,36 @@ def main(args=None):
         parsed_url = parsed_url._replace(netloc=netloc)
         url = parsed_url.geturl()
 
-    if parsed.verbose:
+    if parsed_url.scheme in NOOP_PROTOCOLS:
+        if parsed.verbose:
+            print(f"Skipping because protocol {parsed_url.scheme} is noop")
+        sys.exit(0)
+
+    if parsed.conn_only:
+        host = parsed_url.hostname
+        port = parsed_url.port or DEFAULT_PORTS.get(parsed_url.scheme, None)
+        sock = (host, port)
+        if parsed.verbose:
+            print(
+                f"Testing {host}:{port} for connection with timeout {parsed.timeout}..."
+            )
+    elif parsed.verbose:
         print(f"Testing {url} for {ok_codes!r} with timeout {parsed.timeout}...")
 
     start_time = time.time()
 
     last_fail = ""
     while True:
         try:
-            with urllib.request.urlopen(url, timeout=5) as resp:
-                if resp.code in ok_codes:
-                    sys.exit(0)
-                last_fail = f"HTTP status code: {resp.code}"
+            if parsed.conn_only:
+                with socket.socket() as s:
+                    s.connect(sock)
```

Comment:
I think we want to set the timeout before this line.

```suggestion
                    s.settimeout(5.0)
                    s.connect(sock)
```
https://docs.python.org/3/library/socket.html#socket.socket.settimeout

https://docs.python.org/3/library/socket.html#notes-on-socket-timeouts

Reply:
We also need to catch `ConnectionRefusedError` and retry.